Source/ACE/Entity/WorldObject.cs

```diff
@@ -2411,35 +2440,39 @@ private PhysicsDescriptionFlag SetPhysicsDescriptionFlag()
 
             return physicsDescriptionFlag;
         }
+        
         // todo: return bytes of data for network write ? ?
-        public void SerializePhysicsData(WorldObject wo, BinaryWriter writer)
+        public void SerializePhysicsData(BinaryWriter writer)
         {
             writer.Write((uint)PhysicsDescriptionFlag);
 
             writer.Write((uint)PhysicsState);
 
+            // PhysicsDescriptionFlag.Movement takes priorty over PhysicsDescription.FlagAnimationFrame
+            // If both are set, only Movement is written.
             if ((PhysicsDescriptionFlag & PhysicsDescriptionFlag.Movement) != 0)
             {
                 if (CurrentMotionState != null)
                 {
-                    var movementData = CurrentMotionState.GetPayload(wo.Guid, wo.Sequences);
-                    writer.Write(movementData.Length); // number of bytes in movement object
-                    writer.Write(movementData);
-                    uint autonomous = CurrentMotionState.IsAutonomous ? (ushort)1 : (ushort)0;
-                    writer.Write(autonomous);
+                    var movementData = CurrentMotionState.GetPayload(Guid, Sequences);
+                    if (movementData.Length > 0)
+                    {
```

Comment:
Look at how the client handles this (the pastebin I posted the other day), and verify this is right.  I'm not sure the movementData is being sent correctly.